modules/codemod/lib/v5/renameButtonRefs.ts

```diff
@@ -1,21 +1,44 @@
 import {API, FileInfo, Options, JSXIdentifier} from 'jscodeshift';
 import {getImportRenameMap} from './getImportRenameMap';
 
+const buttonNames = [
+  'Button',
+  'DeleteButton',
+  'deprecated_Button',
+  'DropdownButton',
+  'HighlightButton',
+  'Hyperlink',
+  'IconButton',
+  'OutlineButton',
+  'TextButton',
+  'ToolbarDropdownButton',
+  'ToolbarIconButton',
+];
+
 export default function transformer(file: FileInfo, api: API, options: Options) {
   const j = api.jscodeshift;
 
   const root = j(file.source);
 
-  const {containsCanvasImports} = getImportRenameMap(j, root, '@workday/canvas-kit-react/button');
+  const {containsCanvasImports, importMap} = getImportRenameMap(
+    j,
+    root,
+    '@workday/canvas-kit-react/button'
+  );
 
   if (!containsCanvasImports) {
     return file.source;
   }
 
   root
     .find(j.JSXIdentifier, (value: JSXIdentifier) => value.name === 'buttonRef')
-    .replaceWith(() => {
-      return j.jsxIdentifier('ref');
+    .replaceWith(nodePath => {
+      // only replace via one of our Button imports
+      const elementName = nodePath.parent?.parent?.value?.name?.name;
```

Comment:
This chaining feels tenuous, but that's been my experience with parsing AST, so it's probably fine.

Reply:
The alternative would be to find all `JSXOpeningElement` nodes and filter based on a match, then walk through all `JSXAttributes` looking for the right `JSXIdentifier` and replace the name. This felt simpler. `nodePath.parent` returns `any`, but I didn't bother to set up proper types. Technically those parents and values could be null, so I just added existential operators.